src/services/azblob/backend.rs

```diff
@@ -388,11 +411,27 @@ impl Builder for AzblobBuilder {
             signer: Arc::new(signer),
             container: self.container.clone(),
             client,
-            _account_name: mem::take(&mut self.account_name).unwrap_or_default(),
+            _account_name: account_name.unwrap_or_else(String::new),
         })
     }
 }
 
+fn infer_storage_name_from_endpoint(endpoint: &str) -> Option<String> {
```

Comment:
split is good enough, we can return error if we are meeting invalid input.

Reply:
It seems it is better to add an invalid input check on the endpoint level. This could be a follow-up.